Formula/opentsdb.rb

```diff
@@ -0,0 +1,147 @@
+             "--disable-silent-rules",
+             "--prefix=#{prefix}",
+             "--mandir=#{man}",
+             "--sysconfdir=#{etc}",
+             "--localstatedir=#{var}/opentsdb"
+      system "make"
+      bin.mkpath
+      system "make", "install-exec-am", "install-data-am"
+    end
+
+    env = {
+      :HBASE_HOME => Formula["hbase"].opt_libexec,
+      :COMPRESSION => (build.with?("lzo") ? "LZO" : "NONE"),
+    }
+    env = Language::Java.java_home_env.merge(env)
+    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
+
+    inreplace pkgshare/"etc/opentsdb/opentsdb.conf", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    etc.install Dir["#{pkgshare}/etc/opentsdb"]
+    (pkgshare/"plugins/.keep").write ""
+  end
+
+  def post_install
+    (var/"cache/opentsdb").mkpath
+
+    system "#{Formula["hbase"].opt_libexec}/bin/start-hbase.sh"
+    sleep 2
+    system "#{bin}/create_table.sh"
+    system "#{Formula["hbase"].opt_libexec}/bin/stop-hbase.sh"
+  end
+
+  def tsdb_args
+    %W[
+      #{opt_bin}/tsdb
+      tsd
+      --config=#{HOMEBREW_PREFIX}/etc/opentsdb/opentsdb.conf
+      --staticroot=#{HOMEBREW_PREFIX}/opt/opentsdb/share/opentsdb/static/
+      --cachedir=#{HOMEBREW_PREFIX}/var/cache/opentsdb
+      --port=4242
+      --zkquorum=localhost:2181
+      --zkbasedir=/hbase
+      --auto-metric
+    ]
+  end
+
+  plist_options :manual => tsdb_args.join(" ")
+
+  def plist
+    program_args = tsdb_args.map { |s| "    <string>#{s}</string>" }.join("\n")
+    <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      <dict>
+        <key>OtherJobEnabled</key>
+        <string>#{Formula["hbase"].plist_name}</string>
+      </dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        #{program_args}
+      </array>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/opentsdb/opentsdb.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/opentsdb/opentsdb.err</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    require "socket"
+
+    cp_r (Formula["hbase"].libexec/"conf"), testpath
+    inreplace (testpath/"conf/hbase-site.xml") do |s|
+      s.gsub! /(hbase.rootdir.*)\n.*/, "\\1\n<value>file://#{testpath}/hbase</value>"
+      s.gsub! /(hbase.zookeeper.property.dataDir.*)\n.*/, "\\1\n<value>#{testpath}/zookeeper</value>"
+    end
+
+    ENV["HBASE_LOG_DIR"]  = (testpath/"logs")
+    ENV["HBASE_CONF_DIR"] = (testpath/"conf")
+    ENV["HBASE_PID_DIR"]  = (testpath/"pid")
+
+    system "#{Formula["hbase"].opt_bin}/start-hbase.sh"
+    sleep 2
+
+    system "#{opt_bin}/create_table.sh"
+
+    pid = fork do
+      exec tsdb_args.join(" ")
```

Comment:
I'm pretty sure this should be `exec(*tsdb_args)` instead, as we want to pass the arguments directly to the executable (the one in the first array element) without involving an intermediate shell.